pegasus_wrapper/workflow.py

```diff
@@ -277,6 +324,159 @@ def run_python_on_parameters(
         logging.info("Scheduled Python job %s", job_name)
         return dependency_node
 
+    def run_python_on_args(
```

Comment:
How much of this code overlaps with `run_python_on_parameters`? Can we factor out into something like a `run_python` function?

Reply:
I'll look into this refactor as I think enough code does overlap to make it worth it.

Reply:
What was the result of this exploration?

Reply:
Did a refactor so that the script which creates the python job is an internal function which the exposed APIs then just call the internal function with a flag indicating if the arguments provided are command-line arguments rather than a parameters file.